[AMD] catch exception when other processes belong to other users

xw285cornell opened the pull request:
Summary:
It is a long known pain point that if other users are running things, the call of `torch.cuda.memory.list_gpu_processes()` will error out:
```
  torch.cuda.memory.list_gpu_processes()
  File "torch/cuda/memory.py", line 647, in list_gpu_processes
    procs = amdsmi.amdsmi_get_gpu_process_list(handle)  # type: ignore[attr-defined]
  File "amdsmi/py_interface/amdsmi_interface.py", line 1946, in amdsmi_get_gpu_process_list
    _check_res(
  File "amdsmi/py_interface/amdsmi_interface.py", line 510, in _check_res
    raise AmdSmiLibraryException(ret_code)
amdsmi.py_interface.amdsmi_exception.AmdSmiLibraryException: Error code:
	10 | AMDSMI_STATUS_NO_PERM - Permission Denied

```

So just catch this error

Test Plan: torch.cuda.memory.list_gpu_processes() no longer fails

Differential Revision: D59901053

eqy commented on torch/csrc/Module.cpp:
```diff
@@ -702,10 +702,14 @@ PyObject* THPModule_setSDPUseFlash(PyObject* _unused, PyObject* arg) {
 PyObject* THPModule_userEnabledFlashSDP(PyObject* _unused, PyObject* noargs) {
+#ifndef USE_FLASH_ATTENTION
+  Py_RETURN_FALSE;
+#else
```
Are these changes related to this PR?

xw285cornell replied:
oops no good catch, I'll remove them

xw285cornell committed:
[AMD] catch exception when other processes belong to other users (#131018)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/131018

It is a long known pain point that if other users are running things, the call of `torch.cuda.memory.list_gpu_processes()` will error out:
```
  torch.cuda.memory.list_gpu_processes()
  File "torch/cuda/memory.py", line 647, in list_gpu_processes
    procs = amdsmi.amdsmi_get_gpu_process_list(handle)  # type: ignore[attr-defined]
  File "amdsmi/py_interface/amdsmi_interface.py", line 1946, in amdsmi_get_gpu_process_list
    _check_res(
  File "amdsmi/py_interface/amdsmi_interface.py", line 510, in _check_res
    raise AmdSmiLibraryException(ret_code)
amdsmi.py_interface.amdsmi_exception.AmdSmiLibraryException: Error code:
	10 | AMDSMI_STATUS_NO_PERM - Permission Denied

```

So just catch this error

Test Plan: torch.cuda.memory.list_gpu_processes() no longer fails

Reviewed By: nmacchioni

Differential Revision: D59901053

clee2000 commented:
@pytorchbot merge -f "merged internally"
